lib/stages/applyGains.cpp

```diff
@@ -381,6 +325,53 @@ void applyGains::apply_thread() {
 }
 
 
+void applyGains::fetch_thread() {
+
+    auto& dm = datasetManager::instance();
+
+    while (!stop_thread) {
+
+        // Wait until we've start receiving data before fetching any updates. By
+        // waiting until then before reading the gains we can test for
+        // consistency with the incoming data
+        if (!started) {
+            continue;
+        }
+
+        // Block for an update, and check if we should just exit
+        auto update = update_fetch_queue.get();
+        if (!update)
+            break;
+
+        auto [update_id, transition_interval, new_ts] = *update;
+
+        std::optional<applyGains::GainData> gain_data;
+        if (read_from_file) {
+            DEBUG("Reading gains from file...");
+            gain_data = read_gain_file(update_id);
+        } else {
+            DEBUG("Fetching gains...");
+            gain_data = fetch_gains(update_id);
+        }
+        if (!gain_data) {
+            WARN("Ignoring gain update with update_id {:s}.", update_id);
+            continue;
+        }
+
+        // update gains
+        state_id_t state_id = dm.create_state<gainState>(update_id, transition_interval).first;
+        GainUpdate gain_update{std::move(gain_data.value()), transition_interval, state_id};
+
+        {
+            // Lock mutex exclusively while we update FIFO
+            std::lock_guard<std::shared_mutex> lock(gain_mtx);
+            gains_fifo.insert(double_to_ts(new_ts), std::move(gain_update));
+        }
+        INFO("Updated gains to {:s}.", update_id);
```

Comment:
I guess this one is okay to stay here, even if it's every 10s?

Reply:
Gah. Dang. Didn't see that comment. Given we're not going to 10s gains yet I'll leave it in for the moment. It's very likely we'll need to do some rewriting of this stage anyway for that.